classes/SpecificPriceRule.php

```diff
@@ -275,14 +275,16 @@ public function getAffectedProducts($products = false)
         } else {
             // All products without conditions
             if ($products && count($products)) {
-                $query = new DbQuery();
-                $query->select('p.`id_product`')
-                    ->select('NULL as `id_product_attribute`')
-                    ->from('product', 'p')
-                    ->leftJoin('product_shop', 'ps', 'p.`id_product` = ps.`id_product`')
-                    ->where('ps.id_shop = ' . (int) $current_shop_id);
-                $query->where('p.`id_product` IN (' . implode(', ', array_map('intval', $products)) . ')');
-                $result = Db::getInstance()->executeS($query);
+                if (!SpecificPrice::getByProductId(0, false, false, (int) $this->id)) {
+                    $query = new DbQuery();
+                    $query->select('p.`id_product`')
+                        ->select('NULL as `id_product_attribute`')
+                        ->from('product', 'p')
+                        ->leftJoin('product_shop', 'ps', 'p.`id_product` = ps.`id_product`')
+                        ->where('ps.id_shop = ' . (int) $current_shop_id);
+                    $query->where('p.`id_product` IN (' . implode(', ', array_map('intval', $products)) . ')');
+                    $result = Db::getInstance()->executeS($query);
+                }
```

Comment:
```suggestion
                } else {
                    $result = [['id_product' => 0, 'id_product_attribute' => null]];
                    $result = [['id_product' => 0, 'id_product_attribute' => null]];
                }
```

Reply:
@Progi1984 This is not necessary. The `$products` param is used (according to the method description) to filter the products affected by the `SpecificPriceRule`. So, unless an array(0) is passed as value of the `$products` params, doesn't have sense to return an affected product (id_product => 0) this value is not present on the `$products` params.
Additionally, this method `getAffectedProducts` is only called specifying params on the apply function of the same `SpecificPriceRule` class. It is call previous to creating of specific price to the products list returned. If an specific price is already created with `id_product = 0`, then it is not necessary to create again any other rule for a real product (because an specific price with `id_product = 0` is already affecting all products) or for `id_product = 0` (because is already created). This is the main issue that this PR fixs. In ALL the remaining PrestaShop code, the method `getAffectedProducts` is called without params.